src/helpers.c

```diff
@@ -152,12 +152,15 @@ static int ensure_root_owned_dir(const char *dirname)
 	/* Oops, not owned by root or
 	 * not a directory or wrong perms
 	 */
-	ret = swupd_rm(state_dir);
-	if (ret) {
+	swupd_rm(state_dir);
```

Comment:
Something I didn't notice before: Shouldn't this be `dirname` instead of assuming the function is being called on `state_dir`? Otherwise why pass anything in at all?